### PR TITLE
py-diameter-synthesis, py-region-grower: update; py-tns is now py-neurots

### PR DESCRIPTION
**Pull request by adrien-berchet**

### CODE REVIEW

**Comment by adrien-berchet** [comment]
When trying the `py-region-grower/0.3.1` modules with commands suggested by the bot, I get the following exception:
```
>>> import region_grower.synthesize_morphologies
---------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
/gpfs/bbp.cscs.ch/ssd/apps/bsd/2022-01-10/stage_applications/install_gcc-11.2.0-skylake/py-numpy-1.19.5-gjm7kk/lib/python3.9/site-packages/numpy/core/__init__.py in <module>
     21 try:
---> 22     from . import multiarray
     23 except ImportError as exc:
   
/gpfs/bbp.cscs.ch/ssd/apps/bsd/2022-01-10/stage_applications/install_gcc-11.2.0-skylake/py-numpy-1.19.5-gjm7kk/lib/python3.9/site-packages/numpy/core/multiarray.py in <module>
     11
---> 12 from . import overrides
     13 from . import _multiarray_umath
   
/gpfs/bbp.cscs.ch/ssd/apps/bsd/2022-01-10/stage_applications/install_gcc-11.2.0-skylake/py-numpy-1.19.5-gjm7kk/lib/python3.9/site-packages/numpy/core/overrides.py in <module>
      6
----> 7 from numpy.core._multiarray_umath import (
      8     add_docstring, implement_array_function, _get_implementing_args)
   
ModuleNotFoundError: No module named 'numpy.core._multiarray_umath'
   
During handling of the above exception, another exception occurred:
   
ImportError                               Traceback (most recent call last)
/gpfs/bbp.cscs.ch/ssd/apps/bsd/2022-01-10/stage_applications/install_gcc-11.2.0-skylake/py-dask-2021.6.2-mlnkau/lib/python3.9/site-packages/dask/dataframe/__init__.py in <module>
      2     from ..base import compute
----> 3     from . import backends, dispatch, rolling
      4     from .core import (
   
/gpfs/bbp.cscs.ch/ssd/apps/bsd/2022-01-10/stage_applications/install_gcc-11.2.0-skylake/py-dask-2021.6.2-mlnkau/lib/python3.9/site-packages/dask/dataframe/backends.py in <module>
      2
----> 3 import numpy as np
      4 import pandas as pd
   
/gpfs/bbp.cscs.ch/ssd/apps/bsd/2022-01-10/stage_applications/install_gcc-11.2.0-skylake/py-numpy-1.19.5-gjm7kk/lib/python3.9/site-packages/numpy/__init__.py in <module>
    139
--> 140     from . import core
    141     from .core import *
   
/gpfs/bbp.cscs.ch/ssd/apps/bsd/2022-01-10/stage_applications/install_gcc-11.2.0-skylake/py-numpy-1.19.5-gjm7kk/lib/python3.9/site-packages/numpy/core/__init__.py in <module>
     47         __version__, exc)
---> 48     raise ImportError(msg)
     49 finally:
   
ImportError:
   
IMPORTANT: PLEASE READ THIS FOR ADVICE ON HOW TO SOLVE THIS ISSUE!
   
Importing the numpy C-extensions failed. This error can happen for
many reasons, often due to issues with your setup or how NumPy was
installed.
   
We have compiled some common reasons and troubleshooting tips at:
   
    https://numpy.org/devdocs/user/troubleshooting-importerror.html
   
Please note and check the following:
   
  * The Python version is: Python3.7 from "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/external-libraries/2020-02-01/linux-rhel7-x86_64/gcc-8.3.0/python-3.7.4-tfxecymrkn/bin/python3"
  * The NumPy version is: "1.19.5"
   
and make sure that they are the versions you expect.
Please carefully study the documentation linked above for further help.
   
Original error was: No module named 'numpy.core._multiarray_umath'
   
   
The above exception was the direct cause of the following exception:
   
ImportError                               Traceback (most recent call last)
<ipython-input-2-a80c12bfecbd> in <module>
----> 1 import region_grower.synthesize_morphologies
   
/gpfs/bbp.cscs.ch/project/proj82/home/aberchet/codes/region-grower/region_grower/synthesize_morphologies.py in <module>
     13 from typing import Sequence
     14
---> 15 import dask.dataframe as dd
     16 import dask.distributed
     17 import morphio
   
/gpfs/bbp.cscs.ch/ssd/apps/bsd/2022-01-10/stage_applications/install_gcc-11.2.0-skylake/py-dask-2021.6.2-mlnkau/lib/python3.9/site-packages/dask/dataframe/__init__.py in <module>
     55         '  python -m pip install "dask[dataframe]" --upgrade  # or python -m pip install'
     56     )
---> 57     raise ImportError(msg) from e
   
ImportError: Dask dataframe requirements are not installed.
   
Please either conda or pip install as follows:
   
  conda install dask                     # either conda install
  python -m pip install "dask[dataframe]" --upgrade  # or python -m pip install
```

**Comment by matz-e** [comment]
Hi @adrien-berchet,

do you use a clean shell for that? I don't see any issues:
```
Hostname: bbpv2.epfl.ch
User: matwolf

matwolf in bbpv2 in ~ via  v15.3.0 via 🐍 v2.7.5
❯ unset MODULEPATH

matwolf in bbpv2 in ~ via  v15.3.0 via 🐍 v2.7.5
❯ . /gpfs/bbp.cscs.ch/ssd/apps/bsd/pulls/1504/config/modules.sh

matwolf in bbpv2 in ~ via  v15.3.0 via 🐍 v2.7.5
❯ module load unstable

matwolf in bbpv2 in ~ via  v15.3.0 via 🐍 v2.7.5
❯ module load py-region-grower/0.3.1
Autoloading python/3.9.7
Autoloading py-morphio/3.3.2
Autoloading hpe-mpi/2.25.hmpt

matwolf in bbpv2 in ~ via  v15.3.0 via 🐍 v3.9.7
❯ python
Python 3.9.7 (default, Jan 10 2022, 21:17:49)
[GCC 11.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
Could not open PYTHONSTARTUP
FileNotFoundError: [Errno 2] No such file or directory: '/gpfs/bbp.cscs.ch/home/matwolf/.pystartup'
>>> import region_grower.synthesize_morphologies
>>>
```

**Review by matz-e — DISMISSED** [review]

**Comment by adrien-berchet** [comment]
Hi @matz-e 
Indeed, with a `module purge` before the previous command it works as expected, thanks!

File: bluebrain/repo-bluebrain/packages/py-neurots/package.py [hunk 1]
```diff
@@ -13,15 +13,17 @@ class PyNeurots(PythonPackage):
     git      = "git@github.com:BlueBrain/NeuroTS.git"
 
     version('develop', branch='main')
-    version('3.1.0', tag='3.1.0')
-    version('3.0.0', tag='3.0.0')
+    version('3.1.0', tag='3.1.0')  # Breaking change to use diameter-synthesis >= 0.4
+    version('3.0.0', tag='3.0.0')  # First version of NeuroTS
+    version('2.5.0', tag='2.5.0')  # Last version of TNS before renaming
```
Comment (matz-e) [inline bluebrain/repo-bluebrain/packages/py-neurots/package.py]: Thank you for the comments!

**Review by matz-e — DISMISSED** [review]

**Review by matz-e — DISMISSED** [review]

**Comment by adrien-berchet** [comment]
@matz-e I don't get the issue with the commit message and the PR title, how can I fix this?

**Comment by adrien-berchet** [comment]
Hi @matz-e,
I don't get the CI issues.
For the commit message and the PR title, I tried to update the title but the error message is not really helpful. Do you know how can I fix this?
For the other job, it seems there are permission issues on BB5?
Thanks :)

**Comment by matz-e** [comment]
> Hi @matz-e, I don't get the CI issues. For the commit message and the PR title, I tried to update the title but the error message is not really helpful. Do you know how can I fix this? For the other job, it seems there are permission issues on BB5? Thanks :)

I've fixed the PR title, it needs to be written so that it is easier for us to see what has changed. Just starting with `<most-important-package-that-changed>: <description of changes>` should be enough.

About the permissions: for all public packages, please use `https://` urls, otherwise we have permission issues :)

**Comment by adrien-berchet** [comment]
Aaaah ok ok, thank you very much @matz-e !

File: bluebrain/repo-bluebrain/packages/py-region-grower/package.py [hunk 2]
```diff
@@ -7,26 +7,36 @@
...
-    depends_on("py-neuroc@0.2.8:0.999", type=("build", "run"))
-    depends_on("py-neurom@3.0:3.999", type=("build", "run"))
-    depends_on("py-tns@2.5.0:2.999", type=("build", "run"))
+    depends_on(
+        "py-dask+dataframe+distributed@2.15.0:2021.8.0,2021.9.0:",
+        type=("build", "run"),
+        when="@0.3.1",
+    )
+    depends_on("py-dask+dataframe+distributed@2021.9:", type=("build", "run"), when="@0.4.3:")
+    depends_on("py-diameter-synthesis@0.4.1:0.999", type=("build", "run"), when="@0.3.1")
```
Comment (matz-e) [inline bluebrain/repo-bluebrain/packages/py-region-grower/package.py:29]: For future reference: `:0.999` is equivalent to `:0`, we don't need the trailing 9s any longer :)

**Review by matz-e — APPROVED** [review]

File: bluebrain/repo-bluebrain/packages/py-region-grower/package.py (hunk 2 above)
Comment (adrien-berchet) [reply bluebrain/repo-bluebrain/packages/py-region-grower/package.py:29]: Ah yeah, I realized that yesterday and removed a few ones but not all :)
Do you want me to remove them all in this PR?

Comment (matz-e) [reply bluebrain/repo-bluebrain/packages/py-region-grower/package.py:29]: It can wait, next time you do a version bump you can fix more if you want

Comment (adrien-berchet) [reply bluebrain/repo-bluebrain/packages/py-region-grower/package.py:29]: Ok, I will try to remember next time :)
Thank you very much for your help in this PR!

**Pull request merged**